src/test/java/nustracker/logic/commands/EnrollCommandTest.java

```diff
@@ -1,121 +1,267 @@
-    //        // ensures that outOfBoundIndex is still in bounds of address book list
-    //        assertTrue(outOfBoundIndex.getZeroBased() < model.getAddressBook().getStudentList().size());
-    //
-    //        EnrollCommand enrollCommand = new EnrollCommand(outOfBoundIndex,
-    //                  new EnrolledEvents(VALID_EVENTNAME_TEST));
-    //
-    //        assertCommandFailureShownStudentList(enrollCommand, model,
-    //                                             Messages.MESSAGE_INVALID_STUDENT_DISPLAYED_INDEX);
-    //    }
+    @Test
+    public void constructor_nullParameters_throwsNullPointerException() {
+
+        assertThrows(NullPointerException.class, () -> new EnrollCommand(null, null));
+        assertThrows(NullPointerException.class, () -> new EnrollCommand(STUDENTID_ONE, null));
+        assertThrows(NullPointerException.class, () -> new EnrollCommand(null, EVENTNAME_ONE));
+
+    }
+
+    @Test
+    public void execute_enrollStudent_enrollSuccess() {
+        EnrollCommand enrollCommand = new EnrollCommand(STUDENTID_ONE, MATH_OLYMPIAD.getName());
+
+        ModelStubEnrolledEvents compareModel = new ModelStubEnrolledEvents();
+
+        Student currStudent = model.getStudent(STUDENTID_ONE);
+        Event currEvent = model.getEvent(MATH_OLYMPIAD.getName());
+
+        compareModel.addEnrolledEvent(currStudent, currEvent);
+        compareModel.addParticipant(currEvent, STUDENTID_ONE);
+
+        CommandResult actualCommandResult = null;
+
+        try {
+            actualCommandResult = enrollCommand.execute(model, CurrentlyShownList.STUDENTS_LIST);
+        } catch (CommandException e) {
+            fail("Command Exception should not be thrown.");
+        }
+
+        CommandResult expectedCommandResult = new CommandResult(String.format(
+                MESSAGE_ADD_TO_EVENT_SUCCESS,
+                currStudent.getName().toString(),
+                currStudent.getStudentId().getStudentIdString(),
+                currEvent.getName().getEventName()));
+
+        assertEquals(expectedCommandResult, actualCommandResult);
+
+        assertEquals(model.getAddressBook(), compareModel.getAddressBook());
+
+    }
+
+    @Test
+    public void execute_enrollStudent_studentAlreadyEnrolled() {
+        EnrollCommand enrollCommand = new EnrollCommand(HANS.getStudentId(), MATH_OLYMPIAD.getName());
+
+        try {
+            enrollCommand.execute(model, CurrentlyShownList.STUDENTS_LIST);
+        } catch (CommandException actualCommandException) {
+
+            CommandException expectedCommandException = new CommandException(String.format(
+                    MESSAGE_STUDENT_ALREADY_ENROLLED,
+                    HANS.getName().toString(),
+                    HANS.getStudentId().getStudentIdString(),
+                    MATH_OLYMPIAD.getName().getEventName()));
+
+            assertEquals(expectedCommandException.toString(), actualCommandException.toString());
+            return;
+        }
```

Comment:
You can use assertThrows to test for exceptions. It takes in the class of exception, the error string and the executable as 3 parameters.

Reply:
Ok I'll try that